cocos2d/core/components/CCSlider.js

```diff
@@ -207,14 +206,24 @@ var Slider = cc.Class({
         if (!this.handle) { return; }
         var handlelocalPos;
         if (this.direction === Direction.Horizontal) {
-            handlelocalPos = cc.p(-this.node.width * this.node.anchorX + this.progress * this.node.width, 0);
+            handlelocalPos = cc.p((this.progress - this.node.anchorX) * this._maxRange * 2, 0)
         }
         else {
-            handlelocalPos = cc.p(0, -this.node.height * this.node.anchorY + this.progress * this.node.height);
+            handlelocalPos = cc.p(0, (this.progress - this.node.anchorY) * this._maxRange * 2);
         }
         var worldSpacePos = this.node.convertToWorldSpaceAR(handlelocalPos);
         this.handle.node.position = this.handle.node.parent.convertToNodeSpaceAR(worldSpacePos);
-    }
+    },
+    
+    _setMaxRange: function() {
+        if (!this._maxRange) {
```

Comment:
_maxRange can only be set once ? 
will _maxRange be updated if I dynamically modify the size of node or handle?

Reply:
No, it's set only once per slider for now, thanks for paying attention to this. Perhaps it'll be more appropriate just to remove this conditional and update _maxRange on every progress update (almost as it is now).

Reply:
Can you update the code? Thank you very much. 👍